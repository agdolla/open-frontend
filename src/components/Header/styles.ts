import styled from 'react-emotion';
import fonts from 'style/fonts';
import { Colors } from 'style/variables/palette';
import teimposText from 'style/fonts/tiemposText';
import { below } from 'style/mixins/breakpoints';
import { tablet } from 'style/variables/breakpoints';
import { gutter, maxContentWidth } from 'style/mixins/spacing';

export const Wrapper = styled.div`
  ${gutter};
  ${maxContentWidth};
`;

export const Content = styled.div`
  display: flex;
  padding-bottom: 20px;

  ${below(tablet)} {
    flex-direction: column;
  }
`;

export const Title = styled.h1`
  ${fonts.s64.bold};
  margin-top: 20px;
  margin-bottom: 12px;
`;

export const Description = styled.p`
  ${teimposText.s18.regular};
  color: ${Colors.BlueText};
  line-height: 1.56;
  padding-right: 40px;
`;

export const TextWrapper = styled.div`
  padding-right: 80px;

  ${below(tablet)} {
    padding-right: 0;
  }
`;

export const ImageWrapper = styled.div`
  ${below(tablet)} {
    text-align: center;
    margin-top: 28px;
    margin-bottom: 12px;
  }
`;