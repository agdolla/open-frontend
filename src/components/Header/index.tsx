import * as React from 'react';
import Animation from 'components/Animation';

import { Wrapper, Content, Title, Description, TextWrapper, ImageWrapper } from './styles';

const Header = () => (
  <Wrapper>
    <Content>
      <TextWrapper>
        <img src="/images/brandmark.svg" alt="postmates" />
        <Title>Postmates <br /> Open Source</Title>
        <Description>
          The monkey-rope is found in all whalers; but it was only in the
          Pequod that the monkey and his holder were ever tied together. This
          improvement...
        </Description>
      </TextWrapper>
      <ImageWrapper>
        <Animation time={60} />
        <Animation time={20} marginTop={-280} />
      </ImageWrapper>
    </Content>
  </Wrapper>
);

export default Header;
