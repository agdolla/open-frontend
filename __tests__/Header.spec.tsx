import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../src/components/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
