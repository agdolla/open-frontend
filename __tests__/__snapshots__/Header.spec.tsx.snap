// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Wrapper>
  <Content>
    <TextWrapper>
      <img
        alt="postmates"
        src="/images/logo.svg"
      />
      <Title>
        Postmates 
        <br />
         Open Source
      </Title>
      <Description>
        We love open-source software. Check out some of the projects we have shared with the community.
      </Description>
    </TextWrapper>
    <ImageWrapper>
      <Animation
        time={60}
      />
      <Animation
        marginTop={-280}
        time={20}
      />
    </ImageWrapper>
  </Content>
</Wrapper>
`;
